CMakeLists.txt

```diff
@@ -143,71 +100,50 @@ add_library(jluna SHARED
     include/gc_sentinel.hpp
 )
 
-set_target_properties(jluna PROPERTIES
-    LINKER_LANGUAGE C
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
-target_compile_features(jluna PUBLIC cxx_std_20)
-
-### C ADAPTER ##
-
-add_library(jluna_c_adapter SHARED
-    .src/c_adapter.hpp
-    .src/c_adapter.cpp
-)
-
-target_include_directories(jluna_c_adapter
-    PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
-    PUBLIC $<BUILD_INTERFACE:${JULIA_INCLUDE}>
-)
-
-target_link_libraries(jluna_c_adapter PUBLIC $<BUILD_INTERFACE:${JULIA_LIB}>)
-
-set_target_properties(jluna_c_adapter PROPERTIES
-    LINKER_LANGUAGE C
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
-
 target_link_libraries(jluna PUBLIC jluna_c_adapter)
+target_include_directories(jluna PRIVATE "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>")
 
-### EXECUTABLES ###
+# ---- HACK: export all symbols on Windows ----
 
-add_executable(JLUNA_TEST
-    .test/main.cpp
-    .test/test.hpp
-)
-target_link_libraries(JLUNA_TEST jluna ${JULIA_LIB})
+# FIXME: Exporting all symbols is bad practice. Fix this using the
+#        GenerateExportHeader CMake module and visibility proprties.
+set_target_properties(jluna jluna_c_adapter PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS YES)
 
-add_executable(JLUNA_BENCHMARK
-    .benchmark/main.cpp
-    .benchmark/benchmark.hpp
-    .benchmark/benchmark_aux.hpp
-)
-target_link_libraries(JLUNA_BENCHMARK jluna ${JULIA_LIB})
+# ---- Install rules ----
 
-### INSTALL ###
-
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/jluna-config.cmake [[
-if (NOT ((CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION STRGREATER_EQUAL "12.0.0") OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION STRGREATER_EQUAL "10.0.0")))
-    message(FATAL_ERROR "Currently, the only supported compilers are G++10, G++11 (recommended) and Clang-12")
+if(NOT CMAKE_SKIP_INSTALL_RULES)
+  include(cmake/install-rules.cmake)
 endif()
-include("${CMAKE_CURRENT_LIST_DIR}/jluna-targets.cmake")
-]])
-
-install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/jluna-config.cmake
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/jluna
-)
 
-install(TARGETS jluna jluna_c_adapter EXPORT jluna_targets)
+# ---- Developer mode ----
 
-install(EXPORT jluna_targets
-    FILE jluna-targets.cmake
-    NAMESPACE jluna::
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/jluna
-)
+if(NOT jluna_DEVELOPER_MODE)
+  return()
+elseif(NOT PROJECT_IS_TOP_LEVEL)
+  message(
+      AUTHOR_WARNING
+      "Developer mode is intended for developers of jluna"
+  )
+endif()
```

Comment:
```suggestion
if (jluna_DEVELOPER_MODE)
    return()
```

Reply:
While jluna is not totally vendoring (`FetchContent`, manual or otherwise) ready even after this PR, this warning here should be left here as technically its the private bits of the project where you can do whatever you want to, like touch `CMAKE_*` variable and do other things that affect the entire build tree. Consumers who are aware can still silence it explicitly with `-Wno-dev`.

Reply:
That is true, we'll leave it then